### PR TITLE
WEBUMENIA-1044 journal parsing (first version)

### PR DESCRIPTION
**Pull request by eronisko**
Submitting what I have so far:
* parsing of the journal XHTML
* Basic display in Backpack

I had to go for XHTML because the TEI format on fromthepage.com most likely does not produce a valid XML.

I'll submit another PR for
* displaying the entries better in Backpack
* adding more properties (e.g. source page IDs)

### CODE REVIEW

File: app/JournalParser.php [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    public function getParsedWeather()
+    {
+        $trimmed = str_replace('***', '', $this->textContent);
+        $trimmed = trim($trimmed);
+        return $trimmed;
+    }
+
+    public function getParsedContent()
+    {
+        return str_replace('---', '\n', $this->ownerDocument->saveHTML($this));
```
[the remark this answers is not in the document]
Comment (eronisko) [reply app/JournalParser.php]: Whoops, forgot about that bug -- fixed.

As for the double hyphens (`--`) I think I'll just fix them manually as I go over the transcriptions.

File: app/JournalParser.php [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    }
+
+    public function getParsedDate()
+    {
+        $trimmed = preg_replace('/\s/', '', $this->textContent);
+        $trimmed = str_replace('.IX.', '.9.', $trimmed);
+        $trimmed = str_replace('.X.',  '.10.', $trimmed);
+        $trimmed = str_replace('.XI.', '.11.', $trimmed);
+
+        return \DateTime::createFromFormat('d.m.Y', $trimmed);
```
[the remark this answers is not in the document]
Comment (eronisko) [reply app/JournalParser.php]: Didn't know about this, thanks.

File: app/Models/JournalEntry.php [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Model;
+use Backpack\CRUD\CrudTrait;
+
+class JournalEntry extends Model
+{
+    use CrudTrait;
```
[the remark this answers is not in the document]
Comment (eronisko) [reply app/Models/JournalEntry.php]: Added.

File: app/JournalParser.php [hunk 4]
```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace App;
```
[the remark this answers is not in the document]
Comment (eronisko) [reply app/JournalParser.php]: If it's OK I'd leave this as is for now. IMHO right now this is not adding much complexity and I'm not entirely settled on how I want to use the parser yet (i.e. maybe it'll be transformed/merged into something else).

File: database/migrations/2019_09_20_112334_create_journal_entries_table.php [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('journal_entries', function (Blueprint $table) {
+            $table->bigIncrements('id');
+            $table->date('written_at')->unique();
+            $table->text('written_at_raw');
```
Comment (igor-kamil) [inline database/migrations/2019_09_20_112334_create_journal_entries_table.php]: any reason to use `TEXT` here?

```suggestion
            $table->string('written_at_raw');
```

(same bellow for `weather` and `weather_raw`)

Comment (eronisko) [reply database/migrations/2019_09_20_112334_create_journal_entries_table.php]: The only reason was that I wanted to make our lives easier by not having to set the column size. Happy to change it. But I'm curious as to why? Is it that we can't set an index on a `TEXT` column?

Comment (igor-kamil) [reply database/migrations/2019_09_20_112334_create_journal_entries_table.php]: https://qr.ae/TWsnZt

> For most types, all the data for each row is stored together. The larger types: blob and text, are stored separately. This means that there is a performance impact.

**Pull request merged**